module/programs/helper.js

```diff
@@ -389,62 +389,127 @@ module.exports = class ProgramsHelper {
+	 * @param {Object} bodyData - body data.
+	 * @param {Object} userDetails - User Details
+	 * @param {Boolean} organizations - If organizations is Present.
    * @returns {JSON} - Added entities data.
    */
 
-  static addEntitiesInScope(programId, entities,tenantData) {
+  static addEntitiesInScope(programId, bodyData,userDetails,organizations) {
     return new Promise(async (resolve, reject) => {
       try {
+        let tenantId = userDetails.tenantAndOrgInfo.tenantId
+				let orgId = userDetails.tenantAndOrgInfo.orgId[0]
+        const ALL_SCOPE_VALUE = messageConstants.common.ALL_SCOPE_VALUE
         let programData = await programsQueries.programDocuments(
           {
             _id: programId,
             scope: { $exists: true },
             isAPrivateProgram: false,
-            tenantId: tenantData.tenantId
+            tenantId: tenantId,
+            orgId: orgId,
           },
-          ['_id', 'scope.entityType']
+          ['_id', 'scope']
         );
 
         if (!(programData.length > 0)) {
           throw {
             message: messageConstants.apiResponses.PROGRAM_NOT_FOUND,
+            status: httpStatusCode.bad_request.status
           };
         }
 
-        let entityIds = [];
-
-        let entitiesData = await entityManagementService.entityDocuments(
-          {
-            _id: { $in: entities },
-            entityType: programData[0].scope.entityType,
-            tenantId: tenantData.tenantId,
-            orgId:{$in:['ALL',tenantData.orgId]}
-          },
-          ['_id']
-        );
+				 // Build the $addToSet updateObject
+         let updateObject = { $addToSet: {} }
+        let validationExcludedEntitiesKeys = []
+        let tenantDetails
+        let adminTenantAdminRole = [messageConstants.common.ADMIN, messageConstants.common.TENANT_ADMIN]
+        if (gen.utils.validateRoles(userDetails.roles,adminTenantAdminRole)) {
+          // Fetch tenant details to validate organization codes
+          tenantDetails = await userService.fetchTenantDetails(tenantId, userDetails.userToken);
+          if (!tenantDetails?.success || !tenantDetails?.data?.meta) {
+            return resolve({
+              message: messageConstants.apiResponses.FAILED_TO_FETCH_TENANT_DETAILS,
+              status: httpStatusCode.bad_request.status
+            })
+          }
+          if (
+            Array.isArray(tenantDetails?.data?.meta?.validationExcludedScopeKeys) &&
+            tenantDetails.data.meta.validationExcludedScopeKeys.length > 0
+          ) {
+            // Fetch tenant details (will include valid org codes & validationExcludedScopeKeys)
+            validationExcludedEntitiesKeys.push(...tenantDetails.data.meta.validationExcludedScopeKeys);
+          }
 
-        if (!entitiesData.success) {
-          throw {
-            message: messageConstants.apiResponses.ENTITIES_NOT_FOUND,
-          };
+          if (gen.utils.convertStringToBoolean(organizations)) {
```

Comment:
@MallanagoudaB please include some comments for this code block for better understanding. After we modified this logic with ALL key consideration this block code seem bit large